### DISCUSSION TITLE
Create a content state

### DISCUSSION THREAD

**Discussion by tomcrane** [digirati-co-uk/iiif-canvas-panel#18] [How Might We]
A component that will generate a content state annotation (and its encoded representation).

This is part of a viewer's bookmarking functionality, typically. But consider the editorial workflow...

I am writing an article and I want to select a part of some IIIF Image to use (as per #7). What widget could I use from my content creation environment (Quire, Wordpress, some CMS) to, _from that environment_, load some IIIF resource, navigate it to the place I want, select a region, and capture the content state. The thing that does this is a viewer, essentially, but used as a selection tool.

Is this just one of our demonstrator viewers but used in a particular way?

**Comment by stephenwf** [comment]
Atlas does have a box-drawing "item type"
Demo: https://atlas-viewer-storybook.netlify.app/?path=/story/annotations--selection-demo

So the pieces would be fairly easy to put together into something like this

**Comment by tomcrane** [comment]
One of the viewers we should make is a minimal thumbs+cp viewer that can appear in a modal popup. 
This is the basis of the content-state capture tool.
We'll limit it to capturing canvases, or rectangular regions of canvases, within a manifest. No ranges, no duration, etc.
Also no searching for the manifest - you need to know the URL.

We can take this further later but that's the basics - it should work nicely in a modal popup, and yield a content state to the caller.
